zio-kafka/src/main/scala/zio/kafka/consumer/internal/ConsumerAccess.scala

```diff
@@ -8,16 +8,18 @@ import zio.kafka.consumer.ConsumerSettings
 import zio.kafka.consumer.internal.ConsumerAccess.ByteArrayKafkaConsumer
 
 import scala.jdk.CollectionConverters._
+import zio.concurrent.ReentrantLock
 
 private[consumer] final class ConsumerAccess(
   private[consumer] val consumer: ByteArrayKafkaConsumer,
-  access: Semaphore
+  access: ReentrantLock
 ) {
+
   def withConsumer[A](f: ByteArrayKafkaConsumer => A): Task[A] =
     withConsumerZIO[Any, A](c => ZIO.attempt(f(c)))
 
   def withConsumerZIO[R, A](f: ByteArrayKafkaConsumer => RIO[R, A]): RIO[R, A] =
-    access.withPermit(withConsumerNoPermit(f))
+    access.lock *> withConsumerNoPermit(f) <* access.unlock
```

Comment:
Shouldn't that be:
```suggestion
    access.lock.zipRight.withConsumerNoPermit(f).ensuring(access.unlock)
```

Reply:
Or perhaps this is better:
```suggestion
    (access.lock *> withConsumerNoPermit(f)).ensuring(access.unlock)
```